extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java

```diff
@@ -1098,49 +1096,37 @@ private static JaxrsProvidersToRegisterBuildItem getFilteredJaxrsProvidersToRegi
     }
 
     /**
-     * @param index the index to use to find the existing {@link Application}.
      * @return the set of classes returned by the methods {@link Application#getClasses()} and
      *         {@link Application#getSingletons()}.
      */
-    private static Set<String> getAllowedClasses(IndexView index) {
-        final Collection<ClassInfo> applications = index.getAllKnownSubclasses(ResteasyDotNames.APPLICATION);
+    private Set<String> getAllowedClasses(ClassInfo jakartaRestApplicationClass) {
         final Set<String> allowedClasses = new HashSet<>();
         Application application;
-        ClassInfo selectedAppClass = null;
-        for (ClassInfo applicationClassInfo : applications) {
-            if (Modifier.isAbstract(applicationClassInfo.flags())) {
-                continue;
-            }
-            if (selectedAppClass != null) {
-                throw new RuntimeException("More than one Application class: " + applications);
-            }
-            selectedAppClass = applicationClassInfo;
-            if (selectedAppClass.annotationsMap().containsKey(ResteasyDotNames.CDI_INJECT)) {
-                throw new RuntimeException(
-                        "Usage of '@Inject' is not allowed in 'jakarta.ws.rs.core.Application' classes. Offending class is '"
-                                + selectedAppClass.name() + "'");
-            }
+        if (jakartaRestApplicationClass.annotationsMap().containsKey(ResteasyDotNames.CDI_INJECT)) {
+            throw new RuntimeException(
+                    "Usage of '@Inject' is not allowed in 'jakarta.ws.rs.core.Application' classes. Offending class is '"
+                            + jakartaRestApplicationClass.name() + "'");
+        }
```

Comment:
Not the point of this PR, but per the spec an application type should be a CDI bean with a `@ApplicationScoped` scope. There could be other technical reasons this is done in Quarkus, but just an FYI.

Reply:
Thanks for bringing that up. I don't remember why we did this, but it's certainly something to follow up on later

Reply:
This was introduced in https://github.com/quarkusio/quarkus/issues/22277. We should definitely revisit that

Reply:
> We should definitely revisit that

Would be good to have GH issue for it ;)

Reply:
https://github.com/quarkusio/quarkus/issues/44218